File: 0001-0500/0487_findMaxConsecutiveOnes.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        left = 0
        zeroCount = 0
        maxLength = 0
        for right, num in enumerate(nums):
            if num != 1:
                zeroCount += 1
                while zeroCount > 1:
                    if nums[left] == 0:
                        zeroCount -= 1
                    left += 1
            maxLength = max(maxLength, right - left + 1)    
        return maxLength

File: 0001-0500/0487_findMaxConsecutiveOnes.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int left = 0;
        int zeroCount = 0;
        int maxLength = 0;

        for (int right = 0; right < nums.length; right++) {
            int num = nums[right];

            if (num != 1) {
                zeroCount++;

                while (zeroCount > 1) {
                    if (nums[left] == 0) {
                        zeroCount--;
                    }

                    left++;
                }
            }

            maxLength = Math.max(maxLength, right - left + 1);
        }
        
        return maxLength;
    }
}
